automol/util/vector.py

```diff
@@ -154,112 +152,101 @@ def unit_perpendicular(
     :type value_if_parallel: Vector
     :rtype: numpy.ndarray
     """
+    xyz1 = np.subtract(xyz1, orig_xyz)
+    xyz2 = np.subtract(xyz2, orig_xyz)
+    xyz3 = np.cross(xyz1, xyz2)
 
-    xyz1 = numpy.subtract(xyz1, orig_xyz)
-    xyz2 = numpy.subtract(xyz2, orig_xyz)
-    xyz3 = numpy.cross(xyz1, xyz2)
-
-    if numpy.linalg.norm(xyz3) > 1e-7:
+    if np.linalg.norm(xyz3) > 1e-7:
         uxyz3 = unit_norm(xyz3)
     else:
         uxyz3 = value_if_parallel
 
     return uxyz3
 
 
-def unit_bisector(xyz1, xyz2, orig_xyz, outer: bool = False) -> Vector:
+def unit_bisector(
+    xyz1: Sequence | np.ndarray,
+    xyz2: Sequence | np.ndarray,
+    orig_xyz: Sequence | np.ndarray,
+    outer: bool = False,
+) -> np.ndarray:
     """Calculate a unit bisector.
 
     :param xyz1: 3D vector
-    :type xyz1: tuple, list, or numpy nd.array
     :param xyz2: 3D vector
-    :type xyz2: tuple, list, or numpy nd.array
-    :param orig_xyz: origin of coordinate system `xyz1` and `xyz2` are in
-    :type orig_xyz: tuple, list, or numpy nd.array
+    :param orig_xyz: Origin of coordinate system `xyz1` and `xyz2` are in
     :param outer: Get the outer, instead of the inner bisector?, defaults to False
-    :type outer: bool, optional
-    :rtype: numpy.ndarray
+    :return: Unit bisector
     """
-
     ang = central_angle(xyz1, orig_xyz, xyz2)
     rot_ = rotator(
         axis=unit_perpendicular(xyz1, xyz2, orig_xyz),
         ang=ang / 2.0,
         orig_xyz=orig_xyz,
     )
-    xyz = unit_norm(numpy.subtract(rot_(xyz1), orig_xyz))
+    xyz = unit_norm(np.subtract(rot_(xyz1), orig_xyz))
     if outer:
-        xyz = numpy.negative(xyz)
+        xyz = np.negative(xyz)
     return xyz
 
 
 def from_internals(
-    dist=0.0,
-    xyz1=(0.0, 0.0, 0.0),
-    ang=0.0,
-    xyz2=(0.0, 0.0, 1.0),
-    dih=0.0,
-    xyz3=(0.0, 1.0, 0.0),
-):
+    dist: float = 0.0,
+    xyz1: Sequence | np.ndarray = (0.0, 0.0, 0.0),
+    ang: float = 0.0,
+    xyz2: Sequence | np.ndarray = (0.0, 0.0, 1.0),
+    dih: float = 0.0,
+    xyz3: Sequence | np.ndarray = (0.0, 1.0, 0.0),
+) -> tuple[float]:
```

Comment:
The input types look good. The return value just needs to be changed to `Vector`.

Reply:
Also, we'll need to `require_vector_like` for the vector inputs.